client/src/main/java/io/pravega/client/stream/Transaction.java

```diff
@@ -70,9 +70,11 @@
      * Send a transaction heartbeat and increase transaction's timeout by lease amount of milliseconds.
      *
      * @param lease Additional amount of time in milliseconds by which to increase transaction's timeout.
+     * @param switchOver Boolean indicating whether the previous successful ping request was performed on a different
+     *                   controller process by the client.
      * @throws PingFailedException Ping failed.
      */
-    void ping(long lease) throws PingFailedException;
+    void ping(long lease, boolean switchOver) throws PingFailedException;
```

Comment:
If client were to send the request to different controller instances, and switch over happens pretty aggressively, should we look to change how we assign responsibility for measuring txn timeouts on controller. 
like post the ping in a special stream rather than via rpc call? this way for a given txn, a single controller instance will always be handling the pings.